src/Data/URI/Host/Gen.purs

```diff
@@ -0,0 +1,15 @@
+module Data.URI.Host.Gen where
+
+import Prelude
+
+import Control.Monad.Gen as Gen
+import Data.String as S
+import Data.URI.Host (Host(..))
+
+genIPv4 :: forall m. Gen.MonadGen m => m Host
+genIPv4 = do
```

Comment:
This will not generate ips like `123.001.123.123` though, because of the `001` in there.